escrow/contracts/Escrow.cdc

```diff
@@ -0,0 +1,135 @@
+                panic("Entry already exists for this NFT ID in the leaderboard")
+            }
+
+            // Create the entry and add it to the entries map
+            let entry <- create LeaderboardEntry(
+                nftID: nftID,
+                ownerAddress: ownerAddress,
+                nft: <-nft
+            )
+
+            self.entries[nftID] <-! entry
+
+            emit NFTDeposited(leaderboardName: leaderboardName, nftID: nftID, owner: ownerAddress)
+        }
+
+        // Destructor for Leaderboard resource.
+        destroy() {
+            destroy self.entries
+        }
+
+        init(name: String, nftType: Type) {
+            self.name = name
+            self.nftType = nftType
+            self.entries <- {} as @{UInt64: LeaderboardEntry}
+        }
+    }
+
+    // The resource representing an NFT entry in a leaderboard.
+    pub resource LeaderboardEntry {
+        pub let nftID: UInt64
+        pub let ownerAddress: Address
+        pub let nft: @NonFungibleToken.NFT
+
+        // Destroys the NFT.
+        destroy() {
+            destroy self.nft
+        }
+
+        init(nftID: UInt64, ownerAddress: Address, nft: @NonFungibleToken.NFT) {
+            self.nftID = nftID
+            self.ownerAddress = ownerAddress
+            self.nft <- nft
+        }
+    }
+
+    // An interface containing the Admin function that gets leaderboards by name.
+    pub resource interface IAdmin {
+        pub fun getLeaderboard(name: String): &Leaderboard?
+    }
+
+    // The resource representing an admin.
+    pub resource Admin: IAdmin {
+        // A dictionary holding leaderboards.
+        pub var leaderboards: @{String: Leaderboard}
+
+        // Creates a new leaderboard and stores it.
+        pub fun createLeaderboard(name: String, nftType: Type) {
+            // Create a new Leaderboard resource.
+            let newLeaderboard <- create Leaderboard(name: name, nftType: nftType)
+
+            // Store the leaderboard for future access.
+            self.leaderboards[name] <-! newLeaderboard
+
+            // Emit the event.
+            emit LeaderboardCreated(name: name)
+        }
+
+        // Returns a reference to the leaderboard with the given name.
+        pub fun getLeaderboard(name: String): &Leaderboard? {
+            return &self.leaderboards[name] as &Leaderboard?
+        }
+
+        // Destructor for Admin resource.
+        destroy() {
+            destroy self.leaderboards
+        }
+
+        init() {
+            self.leaderboards <- {}
+        }
+    }
+
+    init() {
+        self.AdminStoragePath = /storage/EscrowAdmin
+        let admin <- create Admin()
+        self.account.save(<-admin, to: self.AdminStoragePath)
+
+        self.AdminPublicPath = /public/AdminPublic
+
+        self.account.link<&Admin{IAdmin}>(/public/AdminPublic, target: self.AdminStoragePath)
+    }
+
+    // Provides access to the Admin resource.
+    access(self) fun getAdmin(): &Admin {
```

Comment:
what's the use of this function? It can only be called by this contract. it has a access(self) modifier